crates/typst-cli/src/main.rs

```diff
@@ -30,7 +32,32 @@ thread_local! {
 }
 
 /// The parsed commandline arguments.
-static ARGS: Lazy<CliArguments> = Lazy::new(CliArguments::parse);
+static ARGS: Lazy<CliArguments> = Lazy::new(|| {
+    let args = CliArguments::parse();
+
+    // Validate the combination of input, output, and makefile_deps. There may
+    // be a more elegant way to do this once the following issue is addressed:
+    // https://github.com/clap-rs/clap/issues/3008. Don't change this without
+    // ensuring it won't break the use of makefile_deps in compile.rs
+    let (Command::Compile(ref command) | Command::Watch(ref command)) = args.command
+    else {
+        return args;
+    };
+    if command.makefile_deps.is_none()
+        || matches!(
+            (&command.common.input, &command.output),
+            (Input::Path(_), Some(Output::Path(_)))
```

Comment:
I don't think an output path is strictly necessary. Stdout is a problem, but if the path isn't specified, we can use `CompileCommand::output()` to infer it from the input path.

If we still need custom validation logic, it should be moved into `args.rs`.

Reply:
Ok, I've switched to using `CompileCommand::output()` and I've removed the validation logic in favour of just reporting an error in `write_make_deps` if there's anything we really can't recover from.